### ISSUE TITLE
`static_model_preferences` doesn't accept class-names as strings, interfering with autoloading

### ISSUE THREAD

**Issue by elia** [solidusio/solidus#4040]
**Solidus Version**:

up to 3.0

**To Reproduce**

1. Create a new store and add `config.static_model_preferences.add('Spree::PaymentMethod::StripeCreditCard', …)` as a static preference, using a string as the class-name
2. Try to start the server

**Current behavior**

The following error is raised:

```
/Users/elia/.asdf/installs/ruby/2.7.2/lib/ruby/gems/2.7.0/gems/solidus_core-2.11.7/lib/spree/preferences/static_model_preferences.rb:12:in `block in initialize': undefined method `defined_preferences' for "Spree::PaymentMethod::StripeCreditCard":String (NoMethodError)
	from /Users/elia/.asdf/installs/ruby/2.7.2/lib/ruby/gems/2.7.0/gems/solidus_core-2.11.7/lib/spree/preferences/static_model_preferences.rb:11:in `each'
	from /Users/elia/.asdf/installs/ruby/2.7.2/lib/ruby/gems/2.7.0/gems/solidus_core-2.11.7/lib/spree/preferences/static_model_preferences.rb:11:in `initialize'
	from /Users/elia/.asdf/installs/ruby/2.7.2/lib/ruby/gems/2.7.0/gems/solidus_core-2.11.7/lib/spree/preferences/static_model_preferences.rb:41:in `new'
	from /Users/elia/.asdf/installs/ruby/2.7.2/lib/ruby/gems/2.7.0/gems/solidus_core-2.11.7/lib/spree/preferences/static_model_preferences.rb:41:in `add'
	from …/my-store/config/initializers/solidus.rb:60:in `block in <main>'
```

**Expected behavior**

Solidus should use `constantize` appropriately to avoid autoloading constants during the initialization.

**Comment by kennyadsl** [comment]
The alternative here is passing the class as a constant? Why did you need to use the string?

**Comment by elia** [comment]
> Why did you need to use the string?

Because calling an autoloadable constant will fix the value to its content and if the class needs to be reloaded (= the content of the constant changes) Ruby will keep around the old incarnation of the constant. Besides during initialization Rails will complain that same reason. Hope this makes sense, happy to add examples if needed 👍

**Comment by kennyadsl** [comment]
@elia 🤯 definitely, some concrete examples would help. I'm not sure to get exactly which kind of code/usage needs this.  Thanks 🙏

**Comment by elia** [comment]
@kennyadsl sure, on one app we have the following inside `config/initializers/solidus.rb`:

```rb
config.static_model_preferences.add(
  Spree::PaymentMethod::StripeCreditCard,
  'stripe_env_credentials',
  secret_key: ENV['STRIPE.SECRET_KEY'],
  publishable_key: ENV['STRIPE.PUBLISHABLE_KEY'],
  server: ENV['RELEASE.STAGE'] == 'production' ? 'production' : 'test',
  test_mode: (ENV['RELEASE.STAGE'] != 'production'),
  stripe_country: 'US',
  v3_elements: false,
  v3_intents: true,
)
```

This autoloads `Spree::PaymentMethod::StripeCreditCard` during initialization, causing the warning and potential problems highlighted above.

**Comment by kennyadsl** [comment]
Ah, got it. Do you think that the proposed PR is enough to fix it?

**Comment by elia** [comment]
@kennyadsl thanks for pointing that out! I left a [review on the PR](https://github.com/solidusio/solidus/pull/4070#pullrequestreview-677340866).

**Comment by waiting-for-dev** [comment]
Closed by #4449
